src/ucp/core/ucp_worker.c

```diff
@@ -2440,42 +2436,49 @@ ucp_worker_discard_uct_ep_flush_comp(uct_completion_t *self,
 static ucs_status_t
 ucp_worker_discard_uct_ep_pending_cb(uct_pending_req_t *self)
 {
-    uct_worker_cb_id_t cb_id = UCS_CALLBACKQ_ID_NULL;
     ucp_request_t *req       = ucs_container_of(self, ucp_request_t, send.uct);
     uct_ep_h uct_ep          = req->send.discard_uct_ep.uct_ep;
-    ucp_worker_h worker      = req->send.discard_uct_ep.ucp_worker;
     ucs_status_t status;
 
     status = uct_ep_flush(uct_ep, req->send.discard_uct_ep.ep_flush_flags,
                           &req->send.state.uct_comp);
+    if (status == UCS_INPROGRESS) {
+        return UCS_OK;
+    }
+
+    /* UCS_OK is handled here as well */
+    if (status != UCS_ERR_NO_RESOURCE) {
+        ucp_worker_discard_uct_ep_flush_comp(&req->send.state.uct_comp,
+                                             status);
+    }
+    return status;
```

Comment:
if status != NO_RESOURCE should also return UCS_OK

Reply:
if flush returned ok, who will call the completion?

Reply:
> if status != NO_RESOURCE should also return UCS_OK

why? we should put to the pending and wait for the callback invocation



> if flush returned ok, who will call the completion?

it will be done in `ucp_worker_discard_uct_ep_flush_comp()`

Reply:
- if the status is error which is not NO_RESOURCE, the flush failed, we cannot retry
- ok